### DISCUSSION TITLE
AC180

### DISCUSSION THREAD

**Discussion by lardzilla** [Patrick762/hassio-bluetti-bt#61] [Ideas]
I have a AC180 and would love support for it!

I used a fork of Bluetti_ESP32_Bridge and was able to read very basic info from it several months ago and was about to revisit it.

Would be happy to provide whatever info I can get from my AC180

**Comment by Patrick762** [comment]
If you have some of the addresses already, using them in the integration will be very easy.
@stast1 also did a scan on the AC180P https://github.com/Patrick762/hassio-bluetti-bt/issues/45 and it seems like it is using the [V2 protocol](https://github.com/Patrick762/hassio-bluetti-bt/wiki/Protocol-V2).
Could you check if you've gotten similar addresses as in the Wiki?

**Comment by Patrick762** [comment]
I've added basic AC180 support in [0.0.19](https://github.com/Patrick762/hassio-bluetti-bt/releases/tag/0.0.19).
Can you test if the values show up?

**Reply by Fridgeir-9** [reply]
I also use an AC180 and have installed beta 0.0.22. The Bluetti was recognized immediately. But as an entity I only get the percent of the battery and "total_power_generation", the latter of which is empty for me.

no AC / DC consumption or adding power via PV or power outlet

How can I help determine the addresses? I use the current hassio, but I need a little instructions

**Comment by bres55** [comment]
I currently use, bluetti_mqtt, running on PC, and mosquito in HA, I would like to get away from the PC bit, hence I am looking at this.
But these are the only useful registers I have been able to make use of with the AC180.

        #Battery Data
        self.struct.add_uint_field('total_battery_percent', 102) #y

        #Power IO
       # self.struct.add_bool_field('ac_output_on',123) #y?  didn't find this useful 
        self.struct.add_uint_field('dc_output_power', 140) #y
        self.struct.add_uint_field('ac_output_power', 142) #y
        self.struct.add_uint_field('dc_input_power', 144) #y
        self.struct.add_uint_field('ac_input_power', 146) #y  I keep forgetting to test this one.

        # Controls (2000)
        self.struct.add_bool_field('ac_output_on', 2011) #y
        self.struct.add_bool_field('dc_output_on', 2012) #y

In that carnation, I had to have this at the end of that file (AC180.py)
    @property
    def writable_ranges(self) -> List[range]:
        return [range(2000, 2225)]

 Haven't been able to follow through, how you do that bit, and if it would be available, with your version.

![image](https://github.com/Patrick762/hassio-bluetti-bt/assets/26557262/1ef91d2c-79ba-4f1a-acbf-80e72c74cbe9)

My main problems are proximity to the ac180 and my HA and PC with bluetooth.
I digress. I'll sort that later, if I can get this to work.
Yes, I got close to nearly draining the battery, hence the need to automate all this.

**Reply by talearns** [reply]
Please provide more details on how you made the new entities appear on your home assistant device. @bres55 

I updated my AC180.py device file with the registers you provided in the comment and restarted the home assistant however I don't see the new entities show up

**Reply by djezlee** [reply]
> Please provide more details on how you made the new entities appear on your home assistant device. @bres55
> 
> I updated my AC180.py device file with the registers you provided in the comment and restarted the home assistant however I don't see the new entities show up

Did you figure it out?  If so, please share.

**Comment by bres55** [comment]
I wrote a bit more here
https://github.com/warhammerkid/bluetti_mqtt/pull/109

**Comment by radfallenn** [comment]
Muito bom ver que podemos controlar via HA e NodeRed, porém a perda de sinal é muito frequente, teria que ficar mais estável. Uso o HA pelo Orange Pi, tem alguma coisa que posso fazer pra melhorar isso?

**Reply by radfallenn** [reply]
It is very good to see that we can control via HA and NodeRed, but the loss of signal is very frequent, it would have to be more stable. I use HA on the Orange Pi, is there anything I can do to improve this?
